promql/functions.go

```diff
@@ -370,7 +370,12 @@ func instantValue(vals []parser.Value, args parser.Expressions, out Vector, isRa
 		if !isRate || ss[1].F >= ss[0].F {
 			// Gauge or counter without reset.
 			resultSample.F = ss[1].F - ss[0].F
+		} else if math.IsNaN(ss[0].F) {
+			// We don't need to do anything extra to handle the case where ss[1].F is a NaN, but we do need to check
+			// ss[0].F in the case where we don't do the subtraction operation above.
+			resultSample.F = math.NaN()
 		}
+
```

Comment:
Thanks for catching this. I think we can solve this more elegantly. The problem is that the last PR changed the logic from asking for a decrease in the value to confirming that we have an increase or the same value. Because NaN always says "no" to both, we get the inversion if NaN is involved.

So how about the following:

```suggestion
		if !isRate || !ss[1].F < ss[0].F {
			// Gauge or counter without reset, including NaN for either sample.
			resultSample.F = ss[1].F - ss[0].F
		}
```

Reply:
Makes sense to me, I've changed this in 337cfc6.